src/test/java/com/puppycrawl/tools/checkstyle/AstTreeStringPrinterTest.java

```diff
@@ -70,6 +73,20 @@ public void testParseFile() throws Exception {
                 JavaParser.Options.WITHOUT_COMMENTS);
     }
 
+    @Test
+    public void testPrintBranch() throws Exception {
+        final DetailAST ast = JavaParser.parseFile(
+            new File(getPath("InputAstTreeStringPrinterPrintBranch.java")),
+            JavaParser.Options.WITHOUT_COMMENTS);
+        final String eol = System.lineSeparator();
+        final String expected = "CLASS_DEF -> CLASS_DEF [3:0]" + eol
+                + "|--MODIFIERS -> MODIFIERS [3:0]" + eol
+                + "|   `--LITERAL_PUBLIC -> public [3:0]" + eol;
```

Comment:
as we already merged your first commit, can you move your good EOL function to some test util and reuse it here ?

not a blocker point, but would be nice.

Reply:
Done.
I moved it to AbstractPathTestSupport class and refactored test code